### DISCUSSION TITLE
Meross / Refoss

### DISCUSSION THREAD

**Discussion by donparlor** [krahabb/meross_lan#157] [Q&A] [answered]
Hello @krahabb,

In your hardware compatibility list, you mention MSG100: Smart Wi-Fi Garage Door Opener. While looking on Amazon to buy MSG100, I can only find [RSG100](https://www.amazon.com/Refoss-Garage-Control-Compatible-Assistant/dp/B07PM59X4J/ref=sr_1_6?keywords=refoss+garage+door+opener&qid=1642375375&sprefix=refoss+garage%2Caps%2C96&sr=8-6). 

It seems Meross/Refoss devices are designed the same way, and can use the same app. 

Are you confident your add-on could see and detect Refoss devices?

Here is what some customers said in Amazon reviews about compatibility. 
![Capture d’écran, le 2022-01-16 à 18 20 12](https://user-images.githubusercontent.com/97048708/149682366-f36da489-b26a-4a84-ac61-1550d4f924aa.png)

Thank you!

Vincent

**Comment by CTJohnK** [comment, marked as answer]
I don't know for certain, but would expect it to work.  I have 8  Refoss and 8 Meross smart plugs and both brand's smart wall switches.  All are working with the Meross app and with Meross LAN.

**Reply by donparlor** [reply]
I just ordered RSG100, I'll give it a try and update you soon.

Thanks @CTJohnK

**Reply by krahabb** [reply]
I'm pretty sure too: if they work on the meross app they should work on meross_lan too (given that the 'original' meross device is corretcly implemented/supported)

**Comment by donparlor** [comment]
I confirm, it works! I just received my Refoss RSG100, installed it, added to Meross App, and it appeared in HA. Thank you @CTJohnK and @krahabb!

**Comment by pedrompamaral** [comment]
Hello.
I bought a Refoss product (MSS425F) and it worked with no problems with this integration.

@krahabb I have a firmware update on my mrs100 sutter switches for some time. I'm afraid they can interfere with your integration. Do you have any ideia if there is any problema with firmware updates at this point? Thanks

**Reply by krahabb** [reply]
Hello @pedrompamaral ,
Nice to have you back! By the look of it, latest firmware updates were only closing down on the key hacking so, if you have your devices keys that should be no issue at all.
The latest 2.5.X releases bring some helpful improvements in key configuration so, if you're still relying on the old null key hack configuration, you can now easily recover the 'official' key for your devices directly from the meross_lan configuration UI

**Reply by pedrompamaral** [reply]
Hello @krahabb 
Waw. I didn't even realised that. I've been making all the integration updates, but still didn't even explored the new features. 😅
All i know now is that the last device i bought, the integration in HA was almost fully automatic. Impressive... 👌
Thank you for your great work. 🙂
